apps/protocol-frontend/src/components/ContributionsTable.tsx

```diff
@@ -311,94 +260,110 @@ const ContributionsTable = ({
           </HStack>
         ),
       },
-    ]);
-  };
+    ];
+  }, []);
 
-  const {
-    getTableProps,
-    getTableBodyProps,
-    headerGroups,
-    rows,
-    state: { globalFilter, selectedRowIds },
-    preGlobalFilteredRows,
-    setGlobalFilter,
-    selectedFlatRows,
-    prepareRow,
-    toggleAllRowsSelected,
-  } = useTable(
-    { columns, data, autoResetSelectedRows: false },
-    useFilters,
-    useGlobalFilter,
-    useSortBy,
-    useRowSelect,
-    tableHooks,
-  );
+  const table = useReactTable({
+    data,
+    columns: columnsDefs,
+    state: {
+      sorting,
+      rowSelection: rowSelection,
+      globalFilter,
+    },
+    enableRowSelection: row => row.original.status.name === 'staging',
+    onSortingChange: setSorting,
+    onRowSelectionChange: setRowSelection,
+    onGlobalFilterChange: setGlobalFilter,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+    debugAll: true,
+  });
 
   useEffect(() => {
-    setSelectedContributions(selectedFlatRows);
-  }, [selectedFlatRows, selectedRowIds]);
+    setSelectedContributions(selectedRows);
+  }, [selectedRows, setSelectedContributions]);
 
-  const toggleSelected = () => {
-    toggleAllRowsSelected(false);
+  useEffect(() => {
+    const selectedContributions: UIContribution[] = [];
+    for (const key in rowSelection) {
+      if (rowSelection[key]) {
+        selectedContributions.push(table.getRow(key).original);
+      }
+    }
+    setSelectedRows(selectedContributions);
+  }, [rowSelection, table]);
+
+  const deselectAll = () => {
+    setRowSelection({});
   };
 
   return (
     <Stack>
       <GlobalFilter
-        preGlobalFilteredRows={preGlobalFilteredRows}
+        preGlobalFilteredRows={table.getPreFilteredRowModel().rows}
         globalFilter={globalFilter}
         setGlobalFilter={setGlobalFilter}
       />
       <Box width="100%" maxWidth="100vw" overflowX="auto">
         <InfiniteScroll
-          dataLength={rows.length}
+          dataLength={table.getRowModel().rows.length}
```

Comment:
Pagination does not seem to be working for me

Reply:
Seeing this as well